tests/x11/nis_server.pm

```diff
@@ -14,107 +14,136 @@
+    send_key 'alt-o';                            # other global setting button
+                                                 # NIS Master Server Details Setup
     assert_screen 'nis-server-master-server-detail-setup';
-    send_key 'alt-o';                           # OK
+    send_key 'alt-o';                            # OK
     send_key $cmd{next};
+    # NIS Server Maps Setup
     assert_screen 'nis-server-server-maps-setup';
-    send_key 'tab';                             # jump to map list
-    my $c = 1;                                  # select all maps
+    send_key 'tab';                              # jump to map list
+    my $c = 1;                                   # select all maps
     while ($c <= 11) {
         send_key 'spc';
         send_key 'down';
         $c++;
     }
-    wait_still_screen 4;
-    save_screenshot;
+    assert_screen 'nis-server-server-maps-setup-finished';
     send_key $cmd{next};
-    send_key 'alt-a';                           # add
-    wait_still_screen 4, 4;                     # blinking cursor
-    type_string '255.255.255.0';
+    # NIS Server Query Hosts
+    assert_screen 'nis-server-query-hosts-setup';
+    send_key 'alt-a';                            # add
+    assert_screen 'nis-server-network-conf-popup';
+    type_string $setup_nis_nfs_x11{net_mask};
     send_key 'tab';
-    type_string '10.0.2.0';
-    send_key 'alt-o';                           # OK
-    save_screenshot;
-    send_key 'alt-f';                           # finish
-    assert_screen 'yast2_closed_xterm_visible';
-    mutex_create('nis_ready');                  # setup is done client can connect
-    type_string "yast2 nfs_server\n";
+    type_string $setup_nis_nfs_x11{net_address};
+    assert_screen 'nis-server-edit-netmask-network';
+    wait_screen_change { send_key 'alt-o' };     # OK
+    assert_screen 'nis-server-query-hosts-setup-finished';
+    send_key 'alt-f';                            # finish
+}
+
+sub nfs_server_configuration {
+    # NFS Server Configuration
     assert_screen 'nfs-server-configuration';
-    send_key 'alt-f';                           # open port in firewall
-    send_key 'alt-m';                           # NFSv4 domain name field
-    type_string 'nfs.openqa.suse.de';
-    wait_still_screen 4, 4;                     # blinking cursor
-    send_key 'alt-s';                           # start nfs server
-    wait_still_screen 4, 4;                     # blinking cursor
-    send_key 'alt-n';                           # next / OK
+    send_key 'alt-f';                            # open port in firewall
+    send_key 'alt-m';                            # NFSv4 domain name field
+    type_string $setup_nis_nfs_x11{nfs_domain};
+    wait_still_screen 4, 4;                      # blinking cursor
+    send_key 'alt-s';                            # start nfs server
+    wait_still_screen 4, 4;                      # blinking cursor
+    send_key 'alt-n';                            # next / OK
```

Comment:
Similarly to the point mentioned above, this looks like unstable code, we do tons of actions and sync only on some pop-ups in between. Your changes make test more readable, but initial problem is with the stability. Patching just one place will bring this test back to us and it will be never ending story (like with yast2_gui and yast2_ncurses).

Reply:
Sure, I can rework this part as well. It have not noticed any problem in the past with this part, so left it open. I will update the code. Thanks